src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,184 @@
+
+protected:
+    struct PhysicalSet {
+        std::unordered_set<int> unusedIndexes;
+        int maxIndexValue;
+
+        PhysicalSet(int maxIndexValue)
+                : maxIndexValue(maxIndexValue) {
+            for (int i = 0; i <= maxIndexValue; ++i)
+                unusedIndexes.insert(i);
+        }
+
+        void setAsUsed(int regIdx) {
+            assert(regIdx <= maxIndexValue && regIdx >= 0);
+            auto it = unusedIndexes.find(regIdx);
+            assert(it != unusedIndexes.end());
+            unusedIndexes.erase(it);
+        }
+
+        void setAsUnused(int regIdx) {
+            assert(regIdx <= maxIndexValue && regIdx >= 0);
+            unusedIndexes.insert(regIdx);
+        }
+
+        int getUnused() {
+            assert(!unusedIndexes.empty());
+            return *unusedIndexes.begin();
+        }
+
+        void exclude(Xbyak::Reg reg) {
+            unusedIndexes.erase(reg.getIdx());
+        }
+    };
+
+    RegistersPool(std::initializer_list<Xbyak::Reg> regsToExclude, int simdRegistersNumber)
+            : simdSet(simdRegistersNumber-1) {
+        using namespace Xbyak;
+        for (auto& reg : regsToExclude) {
+            if (reg.isYMM() || reg.isXMM() || reg.isZMM()) {
+                simdSet.exclude(reg);
+            } else if (reg.isOPMASK()) {
+                excludeOpmask(reg);
+            } else if (reg.isREG()) {
+                generalSet.exclude(reg);
+            }
+        }
+    }
+
+    template<class T>
+    std::tuple<T, std::function<void()>> getFree() {
+        using namespace Xbyak;
+        static_assert(std::is_same<T, Ymm>::value || std::is_same<T, Zmm>::value || std::is_same<T, Xmm>::value ||
+                      std::is_same<T, Reg64>::value || std::is_same<T, Reg32>::value || std::is_same<T, Opmask>::value,
+                      "The AvxRegistersGuardPool::getFree() method unsupported type");
+        if (std::is_same<T, Xmm>::value || std::is_same<T, Ymm>::value || std::is_same<T, Zmm>::value) {
+            auto idx = simdSet.getUnused();
+            simdSet.setAsUsed(idx);
+            return {T(idx), [this, idx](){ simdSet.setAsUnused(idx); }};
+        } else if (std::is_same<T, Reg64>::value || std::is_same<T, Reg32>::value) {
+            auto idx = generalSet.getUnused();
+            generalSet.setAsUsed(idx);
+            return {T(idx), [this, idx](){ generalSet.setAsUnused(idx); }};
+        } else if (std::is_same<T, Opmask>::value) {
+            int idx;
+            std::function<void()> onDestroy;
+            std::tie(idx, onDestroy) = getFreeOpmask();
+            return {T(idx), onDestroy};
+        }
+    }
+
+    virtual std::tuple<int, std::function<void()>> getFreeOpmask() {
+        throw std::runtime_error("getFreeOpmask: The Opmask is not supported in current instruction set");
+    }
+
+    virtual void excludeOpmask(Xbyak::Reg reg) {
+    }
+
+    PhysicalSet generalSet {15};
+    PhysicalSet simdSet; // Xmm, Ymm, Zmm
```

Comment:
This comment with `// Xmm, Ymm, Zmm`, because in future it could be extended for old SSE with MMX, but comment do not describe it.
You have a good name `simdSet`, I think it is enoght

Reply:
Removed